crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -100,15 +122,21 @@ impl AccountBuilder {
         #[cfg(all(not(feature = "testing"), not(test)))]
         let vault = AssetVault::default();
 
-        let (code, storage) =
-            Account::initialize_from_components(self.account_type, &self.components).map_err(
-                |err| {
-                    AccountError::BuildError(
-                        "account components failed to build".into(),
-                        Some(Box::new(err)),
-                    )
-                },
-            )?;
+        let auth_component = self
+            .auth_component
+            .take()
+            .ok_or(AccountError::BuildError("auth component must be set".into(), None))?;
+
+        let mut components = vec![auth_component];
+        components.extend(core::mem::take(&mut self.components));
+
+        let (code, storage) = Account::initialize_from_components(self.account_type, &components)
```

Comment:
We should probably add a check inside `Account::initialize_from_components()` to make sure the first component in the list is an auth component.